release/models/platform/openconfig-platform.yang

```diff
@@ -436,6 +443,40 @@ module openconfig-platform {
         corresponding subcomponent reference from the parent
         component.";
     }
+
+    leaf last-switchover-reason {
+      type identityref {
+        base oc-platform-types:COMPONENT_SWITCHOVER_REASON;
+      }
+      description
+        "For components that have redundant roles (e.g. two
+        supervisors in a device, one as master the other as backup),
+        this reports the reason of the last change of the
+        component's role.";
+    }
+
+    leaf last-switchover-time {
+      type oc-yang:date;
+      description
+        "For components that have redundant roles (e.g. two
+        supervisors in a device, one as master the other as backup),
+        this reports the time of the last change of the
+        component's role.";
+    }
```

Comment:
Would this make sense to be placed into a separate `controller-card` container to imply that these settings are exclusive to `CONTROLLER_CARD` type? Maybe I just don't know of any other component types that would use `switchover`.

Reply:
I have no preference. Maybe @robshakir and @aashaikh can give some guidance regarding which of the following is more aligned with the style? 

1. A direct leaf to the `state` container, e.g: `state/last-switchover-reason`, `state/last-switchover-time`
2. Make a new container, e.g.: `state/redundancy/last-switchover-reason`, `state/redundancy/last-switchover-time`
3. Conditional leaf, e.g. only if the component type is controller card.

Reply:
I think this is a judgement call and preference for the community.  Would be good to hear if there are additional comments.  IMO, I like to avoid the use of conditionals in the models, although it is explicitly permitted if needed:  https://www.openconfig.net/docs/style-guide/

Reply:
Agree that conditions add complexity. I'll leave the structure as it is now.